src/sql/workbench/contrib/notebook/browser/calloutDialog.ts

```diff
@@ -0,0 +1,330 @@
+	private readonly linkAddressLabel = localize('callout.linkAddressLabel', "Address");
+	private readonly linkAddressPlaceholder = localize('callout.linkAddressPlaceholder', "Link to an existing file or web page");
+	// Image
+	private readonly locationLabel = localize('callout.locationLabel', "Image location");
+	private readonly localImageLabel = localize('callout.localImageLabel', "This computer");
+	private readonly remoteImageLabel = localize('callout.remoteImageLabel', "Online");
+	private readonly pathInputLabel = localize('callout.pathInputLabel', "Image URL");
+	private readonly pathPlaceholder = localize('callout.pathPlaceholder', "Enter image path");
+	private readonly urlPlaceholder = localize('callout.urlPlaceholder', "Enter image URL");
+	private readonly browseAltText = localize('callout.browseAltText', "Browse");
+	private readonly embedImageLabel = localize('callout.embedImageLabel', "Attach image to notebook");
+
+	constructor(
+		calloutInstance: CalloutStyle,
+		title: string,
+		posX: number,
+		posY: number,
+		dialogXYOffset: IDialogXYOffset,
+		@IPathService private readonly _pathService: IPathService,
+		@IFileDialogService private readonly _fileDialogService: IFileDialogService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IContextViewService private contextViewService: IContextViewService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILogService logService: ILogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super(
+			title,
+			TelemetryKeys.SelectImage,
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			{
+				dialogStyle: 'callout',
+				dialogPosition: 'below',
+				positionX: posX,
+				positionY: posY,
+				dialogXYOffset: dialogXYOffset
+			});
+
+		this._selectionComplete = new Deferred<ICalloutDialogOptions>();
+		this._calloutStyle = calloutInstance;
+	}
+
+	/**
+	 * Opens the dialog and returns a promise for what options the user chooses.
+	 */
+	public open(): Promise<ICalloutDialogOptions> {
+		this.show();
+		return this._selectionComplete.promise;
+	}
+
+	public render() {
+		super.render();
+
+		attachModalDialogStyler(this, this._themeService);
+
+		this._insertButton = this.addFooterButton(this.insertButtonText, () => this.insert(this._calloutStyle));
+		attachButtonStyler(this._insertButton, this._themeService);
+
+		this._cancelButton = this.addFooterButton(this.cancelButtonText, () => this.cancel(), null, true);
+		attachButtonStyler(this._cancelButton, this._themeService);
+
+		this.registerListeners();
+	}
+
+	protected renderBody(container: HTMLElement) {
+
+		if (this._calloutStyle === `${IMAGE}`) {
```

Comment:
Just use the image property directly - no reason to have the string stuff here

Reply:
(same for all the other similar things in this file)

Reply:
I modified how this is being done; however, I did not figure out how to use the image property directly. If, after reviewing my latest changes you think I should rework this, please let me know -- perhaps with an example of what you mean as well.